### ISSUE TITLE
[BUG] all classifiers must have a classes_ attribute

### ISSUE THREAD

**Issue by goastler** [alan-turing-institute/sktime#82]
Classifiers must have a classes_ attribute in order to be used in cross validation of the experiments / orchestration pipeline. The classes_ value is generated by a LabelEncoder which is built into sklearn. The label encoding transforms or inverse transforms class labels from strings to indices and vice versa. Indices are required for predict_proba context.

Personally I don't like this behaviour, classifiers should not be forced into having a specific field. Resolutions are:
- Remove classes_ completely by using a different cross validation system
- Let orchestration / experiment classes add the classes_ attribute to the classifier themselves, though this would cause the unlikely issue where a classifier already has a classes_ attribute which is nothing to do with label encoding.

Can anyone think of any other resolutions?

**Comment by jasonlines** [comment]
scikit-learn convention is for classifiers to have a `classes_` so I think we should follow that if we want to maintain compatibility. E.g. https://scikit-learn.org/stable/developers/contributing.html#specific-models

![image](https://user-images.githubusercontent.com/38794632/59778859-2a698000-92af-11e9-9c64-d4d141bf8dd6.png)

I'd argue that if a classifier doesn't have `classes_` then it should be updated to have it :) we can probably add a utility function if necessary to cover cases where classifiers don't conform, but in general it should probably be enforced through the extension guidelines and unit testing to make sure all estimators have it. Would be a pain to have a classifier that you can't use grid search cv etc with it!

**Comment by TonyBagnall** [comment]
I think we keep classes_ but see if we can unit test to see if it is set properly
